src/main/java/htsjdk/samtools/util/BufferedLineReader.java

```diff
@@ -26,32 +26,36 @@
 import htsjdk.samtools.Defaults;
 
 import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.InputStreamReader;
+import java.io.LineNumberReader;
 import java.nio.charset.Charset;
 
 /**
  * Implementation of LineReader that is a thin wrapper around BufferedReader.  On Linux, this is faster
  * than AsciiLineReaderImpl.  If you use AsciiLineReader rather than this class, it will detect the OS
  * and delegate to the preferred implementation.
- *
- * TODO: Replace this with {@link java.io.LineNumberReader}?
  * 
  * @author alecw@broadinstitute.org
  */
-public class BufferedLineReader implements LineReader {
-
-    private final BufferedReader reader;
-    private int lineNumber = 0;
-    private String peekedLine;
+public class BufferedLineReader extends LineNumberReader implements LineReader {
 
     public BufferedLineReader(final InputStream is) {
         this(is, Defaults.NON_ZERO_BUFFER_SIZE);
     }
 
     public BufferedLineReader(final InputStream is, final int bufferSize) {
-        reader = new BufferedReader(new InputStreamReader(is, Charset.forName("UTF-8")), bufferSize);
+        super(new InputStreamReader(is, Charset.forName("UTF-8")), bufferSize);
+    }
+
+    /**
+     * Returns a {@link BufferedLineReader} that gets its input from a String. No charset conversion
+     * is necessary because the String is in unicode.
+     */
+    public static BufferedLineReader fromString(final String s) {
```

Comment:
Why not make this another constructor instead of a static method?  It seems like it would fit more naturally with the class if you did that.

Reply:
For my point of view, a Reader is usually constructed from a file or stream, so it looks wierd to have a String constructor, which may be confused by a method like this:

```java
public BufferedLineReader(final String pathString) throws IOException {
    this(Files.newInputStream(Paths.get(pathString)));
}
```

If it is not a good reason, let me know to change it ASAP. I will rebase anyway to remove the redundant commit.

Reply:
That's a good point about the path string.  I hadn't thought of that.